ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java

```diff
@@ -217,7 +209,11 @@ public ProcessingLogger getProcessingLogger() {
   public Optional<Materialization> getMaterialization(
       final QueryId queryId,
       final QueryContext.Stacker contextStacker) {
-    return materializationProvider.map(builder -> builder.build(queryId, contextStacker));
+    return materializationProviderBuilder
+            .flatMap(builder -> builder.apply(
+                    this.sharedKafkaStreamsRuntime.getKafkaStreams(),
+                    topology
+            )).map(builder -> builder.build(queryId, contextStacker));
```

Comment:
Is it possible to have a unit test for this issue?  (Asking naively)

Reply:
Yes just finished adding one :)